File: CSharp/Eta/Src/Codec/RmtesInfo.cs
/*|-----------------------------------------------------------------------------
 *|            This source code is provided under the Apache 2.0 license      --
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.  --
 *|                See the project's LICENSE.md for details.                  --
 *|           Copyright (C) 2022 Refinitiv. All rights reserved.            --
 *|-----------------------------------------------------------------------------
 */

using Refinitiv.Eta.Common;

namespace Refinitiv.Eta.Codec
{
    class RmtesInfo
    {
        internal RmtesBuffer StringBuffer { get; set; }
        internal CodecReturnCode CodecReturnCode { get; set; }
        internal RmtesWorkingSet Set { get; set; }      
        internal int Value { get; set; }       
        internal char CharValue { get; set; }
        internal ByteBuffer Iter { get; set; }
        internal ESCReturnCode ESCRetCode { get; set; }

        internal RmtesInfo ReturnUCS2ToUTF8(ByteBuffer data, int value)
        {
            Iter = data;
            Value = value;
            return this;
        }

        internal RmtesInfo ReturnUTF8ToUCS2(int value, char c)
        {
            Value = value;
            CharValue = c;
            return this;
        }

        internal RmtesInfo ReturnControlParse(ESCReturnCode escSuccess, RmtesWorkingSet currentSet, int i, ByteBuffer iIter)
        {
            ESCRetCode = escSuccess;
            Set = currentSet;
            Value = i;
            Iter = iIter;
            return this;
        }
    }
}

File: Java/Eta/Core/src/main/java/com/refinitiv/eta/codec/RmtesInfo.java
/*|-----------------------------------------------------------------------------
 *|            This source code is provided under the Apache 2.0 license
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.
 *|                See the project's LICENSE.md for details.
 *|           Copyright (C) 2019-2022 LSEG. All rights reserved.     
 *|-----------------------------------------------------------------------------
 */

package com.refinitiv.eta.codec;

import java.nio.ByteBuffer;

import com.refinitiv.eta.codec.RmtesBuffer;
import com.refinitiv.eta.codec.CharSet.RmtesWorkingSet;

class RmtesInfo
{
    RmtesBuffer stringBuffer;
    int codecReturnCode;
    RmtesWorkingSet set;
    private int retCode;
    int value;
    private ByteBuffer iter;
    char charValue;
    String output = new String();

    RmtesBuffer getStringBuffer()
    {
        return stringBuffer;
    }

    int getCodecReturnCode()
    {
        return codecReturnCode;
    }

    int getValue()
    {
        return value;
    }

    RmtesWorkingSet getSet()
    {
        return set;
    }

    int getRetCode()
    {
        return retCode;
    }

    char getCharValue()
    {
        return charValue;
    }
     
    RmtesInfo returnUCS2ToUTF8(ByteBuffer data, int _value)
    {
        setIter(data);
        value = _value;
        return this;
    }

    RmtesInfo returnUTF8ToUCS2(int _value, char c)
    {
        value = _value;
        charValue = c;
        return this;
    }

    RmtesInfo returnControlParse(int escSuccess, RmtesWorkingSet currentSet, int i, ByteBuffer iIter)
    {
        setRetCode(escSuccess);
        set = currentSet;
        value = i;
        setIter(iIter);
        return this;
    }

    void setRetCode(int retCode)
    {
        this.retCode = retCode;
    }

    ByteBuffer getIter()
    {
        return iter;
    }

    void setIter(ByteBuffer iter)
    {
        this.iter = iter;
    }
 }